pkg/blockbuilder/blockbuilder.go

```diff
@@ -610,11 +607,16 @@ consumerLoop:
 	}
 
 	var err error
-	numBlocks, err = builder.CompactAndUpload(ctx, b.uploadBlocks)
+	blockMetas, err = builder.CompactAndUpload(ctx, b.uploadBlocks)
 	if err != nil {
 		return state, err
 	}
 
+	prev, curr, next := getBlockCategoryCount(sectionEndTime, blockMetas)
+	b.blockBuilderMetrics.blockCounts.WithLabelValues("previous").Add(float64(prev))
+	b.blockBuilderMetrics.blockCounts.WithLabelValues("current").Add(float64(curr))
+	b.blockBuilderMetrics.blockCounts.WithLabelValues("next").Add(float64(next))
```

Comment:
Have you considered putting these metrics inside `TSDBBuilder.CompactAndUpload` (ie. make them a part of `tsdbMetrics` collection). It feels this would require fewer changes in the overall code, and make "magic knowledge", like 2h block ranges, scoped closer together.

Reply:
That was my first try, but giving the information about current block end to TSDB did not look right. I am not too worried about the number of lines of change here because it's mostly replacing old stuff and only a minor addition of this function.